### PR TITLE
[FLINK-20772][State] Make TtlValueState#update follow the description of interface about null values

### PR DESCRIPTION
**Pull request by Zakelly**
## What is the purpose of the change

Currently, the TtlValueState#update will wrap the user value with timestamp and proxy request to the original value state. It does not check the null value, which violates the description of the base interface. This PR fixes this by enforcing null value check before wrapping.

## Brief change log

  - check the null value before wrapping in TtlValueState#update
 

## Verifying this change

This change added tests and can be verified as follows:
  - Added test case `testValueSetNull` in `TtlStateTestBase.java`, and make the value of `TtlValueStateTestContext` be `Long` to throw exception when serializing null.  

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency):  **no**
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: **no**
  - The serializers: **no**
  - The runtime per-record code paths (performance sensitive): **yes**
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: **no**
  - The S3 file system connector: **no**

## Documentation

  - Does this pull request introduce a new feature? **no**
  - If yes, how is the feature documented? **not applicable**

### CODE REVIEW

**Comment by Zakelly** [comment]
@flinkbot run azure

File: flink-runtime/src/main/java/org/apache/flink/runtime/state/ttl/TtlValueState.java [hunk 1]
```diff
@@ -48,7 +48,11 @@ public T value() throws IOException {
     @Override
     public void update(T value) throws IOException {
         accessCallback.run();
-        original.update(wrapWithTs(value));
+        if (value == null) {
```
Comment (masteryhx) [inline flink-runtime/src/main/java/org/apache/flink/runtime/state/ttl/TtlValueState.java:51]: How about just make it like: `original.update(value == null ? null : wrapWithTs(value));` ?
Just let the original one to process the logic of updating null which should also follow the contract.

File: flink-runtime/src/test/java/org/apache/flink/runtime/state/ttl/TtlValueStateTestContext.java [hunk 2]
```diff
@@ -21,14 +21,14 @@
 import org.apache.flink.api.common.state.State;
 import org.apache.flink.api.common.state.StateDescriptor;
 import org.apache.flink.api.common.state.ValueStateDescriptor;
-import org.apache.flink.api.common.typeutils.base.StringSerializer;
+import org.apache.flink.api.common.typeutils.base.LongSerializer;
 
 /** Test suite for {@link TtlValueState}. */
 class TtlValueStateTestContext
```
Comment (masteryhx) [inline flink-runtime/src/test/java/org/apache/flink/runtime/state/ttl/TtlValueStateTestContext.java:27]: Maybe I missed something.
Why do we have to modify this class ?

**Review by masteryhx — CHANGES_REQUESTED** [review]
Thanks for the pr.
PTAL my comments.

File: flink-runtime/src/test/java/org/apache/flink/runtime/state/ttl/TtlValueStateTestContext.java (hunk 2 above)
Comment (Zakelly) [reply flink-runtime/src/test/java/org/apache/flink/runtime/state/ttl/TtlValueStateTestContext.java:27]: Yes, we need modify this test, since the StringSerializer supports to serialize null value, which hides the original problem.

**Comment by Zakelly** [comment]
@masteryhx Thanks so much for your review! I have update my PR.

**Review by masteryhx — APPROVED** [review]
Thanks for the update.
LGTM.
